### DISCUSSION TITLE
chipsec: Python2 to Python3 conversion/refactoring

### DISCUSSION THREAD

**Discussion by net-wayfarer** [BlackArch/blackarch#3028] [General]
This discussion is a continuation from a [PR that I previously made](https://github.com/BlackArch/blackarch/pull/3027).

Initially I was misunderstanding that `chipsec` package depended on `python2*` and so I decided to go modifying the `PKGBUILD` further to have `makepkg` do the [conversion/refactoring](https://docs.python.org/3/library/2to3.html) during the `prepare() {}` process,
```
$ grep -A 8 prepare PKGBUILD
prepare() {
  cd "$pkgname"
  
  2to3 -w .

  cd "drivers/linux"

  patch -Np0 < "$srcdir/Makefile.patch"
}
```

What actually ended up happening is most probably that the Python files got refactored (warning, huge paste ahead). I haven't experimented with `chipsec_main.py` yet, but am intending to canvass some opinions on this. Would there be any point in pushing a PR with the aforementioned changes even if it may not be useful? I am also partly thinking aloud here.

Full log: __Deleted__ https://paste.rs/jar (Warning: extremely long log)

**Comment by aancw** [comment]
Hey, can you just paste link for full log? It's so very long scroll.

**Reply by net-wayfarer** [reply]
Sorry, I just went and fixed that.

**Reply by aancw** [reply]
Thank for your understanding 🙂

**Comment by Edu4rdSHL** [comment]
Hello, @net-wayfarer, @aancw, hope that you're fine. I see that chipsec is currently being compiled [using Python3](https://github.com/BlackArch/blackarch/blob/220a3b5a3d32f65fedea880b4d9a6eeba51f576a/packages/chipsec/PKGBUILD#L36), what are the advantages of using 2to3 in this case? I think it will probably bring more problems than improvements.

**Reply by net-wayfarer** [reply]
Hi @Edu4rdSHL. Yeah it is compiled in Python3 as I later realised after checking on older prebuilt package that I did; `chipsec-4:1070.7966175`. Hence brought me to the discussion of questioning would there be any merits in using `2to3 -w .` when there appears to be some changes being applied to the Python files.

In either case, I have not tested `chipsec` fully. Regardless if I refactored it or not, as I was primarily concerned with the error status whenever `pacman` invokes `dkms` during system updates, and notably kernel updates. 

I think at this point, there appears to be no real benefit (if any) and am going to leave the `PKGBUILD` unchanged in the Blackarch repo. One good thing that I didn't push any changes and PR :smile: with regards to this.

Thanks for inputs!

**Reply by Edu4rdSHL** [reply]
Hey @net-wayfarer, I agree, we should keep the `PKGBUILD` without any change, it's working fine currently (at least the module installation) so there isn't a real reason to include the change mentioned above.

Thank you too!
